File: vb/phone-numbers-kata/phone-numbers-kata/Validate.vb
Public Class Validate

    Public Shared Function PhoneNumber(telephoneNumber As String) As String
        Dim numberToCheck = telephoneNumber
        numberToCheck = numberToCheck.Replace(" ", "").Replace("-", "").Replace("(", "").Replace(")", "").Replace(",", "")

        If Not numberToCheck.StartsWith("9") = True Then Return "invalid"
        If Not numberToCheck.Length > 10 Then Return "invalid"
        If Not numberToCheck.Length < 12 Then Return "invalid"
        If Not IsNumeric(numberToCheck) Then Return "invalid"

        Return numberToCheck

    End Function

End Class

File: c#/phone-numbers-kata/phone-numbers-kata/Validate.cs
using System;

class Validate
{
    public static string PhoneNumber(string telephoneNumber)
    {
        string numberToCheck = telephoneNumber;
        numberToCheck = numberToCheck.Replace(" ", "").Replace("-", "").Replace("(", "").Replace(")", "").Replace(",", "");

        if (!(numberToCheck.StartsWith("9"))) return "invalid";
        if (!(numberToCheck.Length > 10)) return "invalid";
        if (!(numberToCheck.Length < 12)) return "invalid";
       // if (!Information.IsNumeric(numberToCheck)) return "invalid";

        return numberToCheck;

    }

}